lib/apiservers/engine/backends/eventmonitor.go

```diff
@@ -120,8 +120,21 @@ func (m *PortlayerEventMonitor) Start() error {
 	}
 
 	m.stop = make(chan struct{})
-	go m.monitor()
-
+	go func() {
+		var err error
+		for {
+			select {
+			case <-m.stop:
+				break
+			default:
+				if err = m.monitor(); err != nil {
+					log.Error(err)
+				}
+				log.Debug("Restarting Portlayer event monitor")
+				m.stop = make(chan struct{})
```

Comment:
no - this races with `m.Stop()` closing the channel that was assigned to this previously.
The m.stop channel should still be happily open when `m.monitor` exits.